src/domToStencil.ts

```diff
@@ -0,0 +1,55 @@
+import { VNode, h } from "@stencil/core";
+import { attributesToProps } from "./attributesToProps";
+import { formatAttributes } from "./formatAttributes";
+import { formatTagName } from "./formatTagName";
+
+type StencilNode = VNode | string;
+
+export function domToStencil(nodes: NodeList): StencilNode[] {
+  let result: StencilNode[] = [];
+
+  for (let index = 0, len = nodes.length; index < len; index++) {
+    let node = nodes[index];
+    let current: StencilNode;
+
+    // set the node data given the type
+    switch (node.nodeType) {
+      case 1:
```

Comment:
It looks like there are constants for `nodeType`, which might be clearer than using bare numbers: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType#Constants